### PR TITLE
Add instructions for developing the app

### PR DESCRIPTION
**Pull request by Philipp91**

### CODE REVIEW

**Comment by JetpackDuba** [comment]
Hello! Thanks for your effort.

Aside from the previous comments, I'd add a section mentioning the need of libssh in order to SSH authentication work. I'd link this https://www.libssh.org/ . The libraries need to be installed globally or be in the same directory as the project (when developing) and in the same directory as the main binary when building a release version (for example, in Windows it would be in the same dir as Gitnuro.exe)

**Comment by Philipp91** [comment]
> Aside from the previous comments

Which do you mean? I just see this one. Maybe you need to send out the review comments in GitHub, if you started a review?

> I'd add a section mentioning the need of libssh

Done. But it's still a little wrong.

> for example, in Windows it would be in the same dir as Gitnuro.exe

So whenever you do a debug run on Windows (`./gradlew run` or sth like that in your IDE), it builds an .exe file? Doesn't it just call the JRE's `java` command with a jar file that it built?

On Linux, it builds a jar and runs it, leaving the PWD at the repository's main directory. So what I wrote so far (asking users to put the binary into `./build` is wrong at least for Linux, and probably also not correct for Windows.

Also, if it puts a .exe somewhere on Windows, isn't that typically in a directory that gets nuked on every build, so putting the .dll in there isn't possible?

**Comment by Philipp91** [comment]
Hm, when I open up the Flatpack package for Linux, it doesn't even contain a `libssh` file. Then how does that work?

**Review by JetpackDuba — CHANGES_REQUESTED** [review]

**Comment by JetpackDuba** [comment]
> Which do you mean? I just see this one. Maybe you need to send out the review comments in GitHub, if you started a review?

Oops, my bad there! I'm not very familiar with PR in GitHub.


> Done. But it's still a little wrong.
> 

> So whenever you do a debug run on Windows (`./gradlew run` or sth like that in your IDE), it builds an .exe file? Doesn't it just call the JRE's `java` command with a jar file that it built?
> 
> On Linux, it builds a jar and runs it, leaving the PWD at the repository's main directory. So what I wrote so far (asking users to put the binary into `./build` is wrong at least for Linux, and probably also not correct for Windows.
>
> Also, if it puts a .exe somewhere on Windows, isn't that typically in a directory that gets nuked on every build, so putting the .dll in there isn't possible?

The library should only be with the executable for a release version, not in the build directory. By default Gradle builds a JAR, but by using a Compose task called "createDistributable", it will create a build for the current OS in the expected format, exe for windows and dmg for MacOS (the portable version of windows and the dmg of Mac in the previous releases are an example of that). For development the lib should be at the project's level or, if you prefer, tell  gradle to include it (I don't exactly remember right now how it worked but a quick search can help).

> Hm, when I open up the Flatpack package for Linux, it doesn't even contain a `libssh` file. Then how does that work?

Because it's a new change for v1.2 that hasn't been released yet.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+[Compose Multiplatform IDE Support](https://plugins.jetbrains.com/plugin/16541-compose-multiplatform-ide-support)
+plugin.
+
+## Alternative: Setting up JDK for use on CLI
+
+You don't need this if you only use the JDK installed by IntelliJ IDEA.
+
+Check which Java version this project currently uses (`cat build.gradle.kts | grep jvmTarget`) and install it:
+```bash
+sudo apt-get install openjdk-17-jre openjdk-17-jdk
```
[the remark this answers is not in the document]
Comment (Philipp91) [reply DEVELOPMENT.md:25]: Done. Also adjusted the `grep` above to #78.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Setting up an IDE
+
+If you don't have another preference, the recommendation is to download and install
+[IntelliJ IDEA Community Edition](https://www.jetbrains.com/idea/download/#section=linux)
+(possibly through the JetBrains Toolbox, if you have it already) as well as the
+[Compose Multiplatform IDE Support](https://plugins.jetbrains.com/plugin/16541-compose-multiplatform-ide-support)
+plugin.
+
+## Alternative: Setting up JDK for use on CLI
```
[the remark this answers is not in the document]
Comment (Philipp91) [reply DEVELOPMENT.md:19]: I'll let others add those, as I couldn't test them myself.

**Comment by Philipp91** [comment]
> For development the lib should be at the project's level, in the system path ...

Rephrased to basically say/explain that.

> ... or, if you prefer, tell gradle to include it (I don't exactly remember right now how it worked but a quick search can help).

I won't explain that. If anyone ever needs such a fancy setup, they can figure it out themselves. The intention of this new md file is to help people to get started quickly, not to describe all development environments exhaustively.

**Comment by Philipp91** [comment]
> Because it's a new change for v1.2 that hasn't been released yet.

Oh ok. Then I wonder if v1.2 would even have worked on my machine without #81.

**Comment by JetpackDuba** [comment]
> > ... or, if you prefer, tell gradle to include it (I don't exactly remember right now how it worked but a quick search can help).
> 
> I won't explain that. If anyone ever needs such a fancy setup, they can figure it out themselves. The intention of this new md file is to help people to get started quickly, not to describe all development environments exhaustively.

Yeah it's fine, I didn't mean for you to add it, just as additional information in case you were interested.

**Pull request merged**